Sources/Yams/Constructor.swift

```diff
@@ -0,0 +1,360 @@
+    static func construct(from node: Node) -> Self?
+}
+
+extension Bool: ScalarConstructible {
+    public static func construct(from node: Node) -> Bool? {
+        guard let scalar = node.scalar else { fatalError("Never happen this") }
+        switch scalar.lowercased() {
+        case "true", "yes", "on":
+            return true
+        case "false", "no", "off":
+            return false
+        default:
+            return nil
+        }
+    }
+}
+
+extension Data: ScalarConstructible {
+    public static func construct(from node: Node) -> Data? {
+        guard let scalar = node.scalar else { fatalError("Never happen this") }
+        let data = Data(base64Encoded: scalar, options: .ignoreUnknownCharacters)
+        return data
+    }
+}
+
+extension Date: ScalarConstructible {
+    public static func construct(from node: Node) -> Date? {
+        guard let scalar = node.scalar else { fatalError("Never happen this") }
+
+        let range = NSRange(location: 0, length: scalar.utf16.count)
+        guard let result = timestampPattern.firstMatch(in: scalar, options: [], range: range),
+            result.range.location != NSNotFound else {
+                return nil
+        }
+        #if os(Linux)
+            let components = (1..<result.numberOfRanges).map(result.range(at:)).map(scalar.substring)
+        #else
+            let components = (1..<result.numberOfRanges).map(result.rangeAt).map(scalar.substring)
+        #endif
+
+        var datecomponents = DateComponents()
+        datecomponents.calendar = Calendar(identifier: .gregorian)
+        datecomponents.year = components[0].flatMap { Int($0) }
+        datecomponents.month = components[1].flatMap { Int($0) }
+        datecomponents.day = components[2].flatMap { Int($0) }
+        datecomponents.hour = components[3].flatMap { Int($0) }
+        datecomponents.minute = components[4].flatMap { Int($0) }
+        datecomponents.second = components[5].flatMap { Int($0) }
+        datecomponents.nanosecond = components[6].flatMap {
+            let length = $0.characters.count
+            let nanosecond: Int?
+            if length < 9 {
+                nanosecond = Int($0 + String(repeating: "0", count: 9 - length))
+            } else {
+                nanosecond = Int($0.substring(to: $0.index($0.startIndex, offsetBy: 9)))
+            }
+            return nanosecond
+        }
+        datecomponents.timeZone = {
+            var seconds = 0
+            if let hourInSecond = components[9].flatMap({ Int($0) }).map({ $0 * 60 * 60 }) {
+                seconds += hourInSecond
+            }
+            if let minuteInSecond = components[10].flatMap({ Int($0) }).map({ $0 * 60 }) {
+                seconds += minuteInSecond
+            }
+            if components[8] == "-" { // sign
+                seconds *= -1
+            }
+            return TimeZone(secondsFromGMT: seconds)
+        }()
+        // Using `DateComponents.date` causes crash on Linux
```

Comment:
Oh, no, that's fine. Sorry I've been reviewing on my phone so I didn't look it up.